qcodes/instrument/base.py

```diff
@@ -106,11 +147,26 @@ def add_parameter(self, name: str,
             **kwargs: Constructor arguments for ``parameter_class``.
 
         Raises:
-            KeyError: If this instrument already has a parameter with this
-                name.
+            ValueError: If this instrument already has a parameter with this
```

Comment:
This still raises an KeyError if the parameter is not abstract. Changing the error type is an api break so I think we should stick to the same error.

Reply:
Actually, the docstring is wrong. I am raising the same error when the parameter already exists and its not an abstract parameter: A `KeyError`. I will change the docstring to reflect this.